### DISCUSSION TITLE
Supervision using Yolov7

### DISCUSSION THREAD

**Discussion by JSNN170** [roboflow/supervision#848] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

Hi,
Is it possible to run supervision using Yolov7 ? If so how?

Thanks.

**Comment by SkalskiP** [comment]
You can use Supervision if model is runnable via SDK or Python library. As far as I know YOLOv7 is officially only runnable via script. Is there any official library?

**Comment by onuralpszr** [comment]
Yolov7 does not provide pip but I created a notebook that can detect and I made record video, technically it is loading from pytorch. I created "from_yolov7" but it is exactly same "from_yolov5"

Collab link : https://colab.research.google.com/drive/18rNyTWgBg6dUeYCPdXEg-t1KhyXWFWg7?usp=sharing

@SkalskiP  maybe we should consider to create from_torch.

**Reply by JSNN170** [reply]
Hi @onuralpszr,

Thanks for your answer,

Your script works on Google Colab and now I am trying to run it on my local machine but i get this error message from the hubconf.py file:

```
Exception has occurred: Exception
Cache maybe be out of date, try force_reload=True.
```

It is related to the line :

`model = yolov7()`

I then tried writing the line this way:

`model = torch.hub.load(repo_or_dir='WongKinYiu/yolov7', model='yolov7', force_reload=True)`

But I get the same error message.


I have all the required libraries and cuda available.
Can you help me with this? 
Thanks.

**Reply by onuralpszr** [reply]
You can try to download via "wget" or manually click the link 

`wget https://github.com/WongKinYiu/yolov7/releases/download/v0.1/yolov7.pt`

but when I tested on collab it was downloading fine and locally on my machine was fine as well.

**Reply by haviduck** [reply]
Thank you @onuralpszr
successfully loaded custom yolov7 model using custom and yolov7 from hubconf:
```
import numpy as np
import torch
from  hubconf import yolov7, custom
import supervision as sv
model = custom(path_or_model='weights/rag_version_1.pt')  # custom example

tracker = sv.ByteTrack()
box_annotator = sv.BoundingBoxAnnotator()
label_annotator = sv.LabelAnnotator()
trace_annotator = sv.TraceAnnotator()

def callback(frame: np.ndarray, _: int) -> np.ndarray:
    results = model(frame)
    detections = sv.Detections.from_yolov7(results)
    detections = tracker.update_with_detections(detections)

    labels = [
        f"#{tracker_id} {results.names[class_id]}"
        for class_id, tracker_id
        in zip(detections.class_id, detections.tracker_id)
    ]

    annotated_frame = box_annotator.annotate(
        frame.copy(), detections=detections)
    annotated_frame = label_annotator.annotate(
        annotated_frame, detections=detections, labels=labels)
    return trace_annotator.annotate(
        annotated_frame, detections=detections)


sv.process_video(
    source_path="input.mp4",
    target_path="result.mp4",
    callback=callback
)
```

**Comment by onuralpszr** [comment]
Converting to discussion as Q&A category

**Comment by SkalskiP** [comment]
As I mentioned in my private conversation with @onuralpszr, we are unlikely to add a separate function for YOLOv7. We are now trying to support primarily libraries and not individual models unless we see increased interest from the community.

However, adding a dedicated cookbook - `Using YOLOv7 with Supervision` - would be a good move.
